components/automate-ui/src/app/entities/layout/layout.facade.ts

```diff
@@ -11,63 +11,111 @@ import { MenuItemGroup } from './layout.model';
 import { ShowPageLoading, UpdateSidebarMenuGroups } from './layout.actions';
 
+import { GetProjects } from 'app/entities/projects/project.actions';
+
 // Important! These must match components/automate-ui/src/styles/_variables.scss
 enum Height {
-  Navigation = '70px',
-  Banner = '110px'
+  Navigation = 70,
+  License = 39,
+  ProcessProgressBar = 48,
+  PendingEditsBar = 36
 }
 
 @Injectable({
   providedIn: 'root'
 })
-export class LayoutFacadeService {
-  headerHeight = '70px';
-  contentHeight = `calc(100% - ${this.headerHeight})`;
-  menuGroups$: Observable<MenuItemGroup[]>;
-  showPageLoading$: Observable<boolean>;
-  showLicenseNotification = false;
-  showHeader = true;
-  showSidebar = true;
+export class LayoutFacadeService implements OnDestroy {
+  public layout = {
+    header: {
+      display: true,
+      license: false,
+      navigation: true
+    },
+    sidebar: {
+      display: true,
+      navigation: true
+    },
+    userNotifications: {
+      display: true,
+      pendingEdits: false,
+      updatesProcessing: false
+    }
+  };
+  public contentHeight = `calc(100% - ${Height.Navigation}px)`;
+  public menuGroups$: Observable<MenuItemGroup[]>;
+  public showPageLoading$: Observable<boolean>;
+  private isDestroyed = new Subject<boolean>();
 
   constructor(
     private store: Store<fromLayout.LayoutEntityState>,
     private layoutSidebarService: LayoutSidebarService
   ) {
+    this.store.dispatch(new GetProjects());
     this.menuGroups$ = store.select(sidebarMenuGroups);
     this.showPageLoading$ = store.select(showPageLoading);
 
     store.select(notificationState).subscribe(
       (notifications: Notification[]) => {
-        this.showLicenseNotification =
+        this.layout.header.license =
           notifications &&  notifications.some(n => n.type === Type.license);
-        this.updateContentHeight(
-          this.showLicenseNotification ? Height.Banner : Height.Navigation);
+          if (this.layout.header.license) {
+            this.layout.header.license = true;
+            this.updateDisplay();
+          } else {
+            this.layout.header.license = false;
+            this.updateDisplay();
+          }
       });
+
+      this.updateDisplay();
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  getContentStyle() {
+    return { 'height': this.contentHeight };
+  }
+
+  updateDisplay() {
+    let combinedHeights = 0;
+    if (this.layout.header.navigation) {
+      combinedHeights += Height.Navigation;
+    }
+    if (this.layout.header.license) {
+      combinedHeights += Height.License;
+    }
+    if (this.layout.userNotifications.pendingEdits) {
+      combinedHeights += Height.PendingEditsBar;
+    }
+    if (this.layout.userNotifications.updatesProcessing) {
+      combinedHeights += Height.ProcessProgressBar;
+    }
+    this.contentHeight = `calc(100vh - ${combinedHeights}px)`;
```

Comment:
What is `vh` ?

Reply:
vw = 1% of the viewport's width
vh = 1% of the viewport's height
https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#Relative_length_units

100vh = full height of the browser viewport

Reply:
@msorens vh is view height https://css-tricks.com/fun-viewport-units/